### PR TITLE
Fix documentation warnings

### PR DESCRIPTION
**Pull request by dlrobertson**
There are several warnings that are thrown when running `cargo doc`. Fix
these warnings.

### CODE REVIEW

**Comment by dlrobertson** [comment]
I'm okay with dropping the code block changes due to [rust-lang/rust#38739](https://github.com/rust-lang/rust/issues/38739)

File: src/lib.rs [hunk 1]
```diff
@@ -10,16 +10,14 @@
 //! for writing applications but also to be a toolbox of networking primitives, so
 //! every layer is fully exposed and documented.
 //!
-//! When discussing networking stacks and layering, often the [OSI model][osi] is invoked.
+//! When discussing networking stacks and layering, often the [OSI model](osi) is invoked.
```
Comment (whitequark) [inline src/lib.rs]: Wait, what? How is this legal Markdown?

File: src/phy/mod.rs [hunk 2]
```diff
@@ -17,73 +17,71 @@
 //! An implementation of the [Device](trait.Device.html) trait for a simple hardware
 //! Ethernet controller could look as follows:
 //!
-/*!
-```rust
```
Comment (whitequark) [inline src/phy/mod.rs]: This change makes it a real pain in the ass to update large code blocks.

File: src/lib.rs (hunk 1 above)
Comment (dlrobertson) [reply src/lib.rs]: Its not :(. I was playing around with some stuff and it got lumped in here.

File: src/phy/mod.rs (hunk 2 above)
Comment (dlrobertson) [reply src/phy/mod.rs]: Agreed. If we're okay with some extra warnings, I'd be okay with going back to `/*!`.

**Comment by dlrobertson** [comment]
Updated to use `/*!` for the entire section.

**Pull request merged**